src/cljs/vanilla/update_layout.cljs

```diff
@@ -1,4 +1,17 @@
-(ns vanilla.update-layout)
+(ns vanilla.update-layout
+  (:require
+    [ajax.core :as ajax :refer [GET POST]]))
+
+
+(defn save-layout [layout]
+  ;(prn "saving layout: " (clojure.core/pr-str layout))
+
+  (POST "/save-layout"
+        {:format          (ajax/json-request-format {:keywords? true})
+         :response-format (ajax/json-response-format {:keywords? true})
+         :params          {:widgets (clojure.core/pr-str layout)}       ;convert the whole layout struct to a string to preserve values
+         :handler         #(prn "Layout SAVED!")
```

Comment:
here is where you can dispatch something to re-frame to indicate that the save was successful (a boolean) that will trigger the appearance of a little notification that the save worked. then after 3 seconds we can set that value back to :false and the notification would fade back to invisible.